src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -131,6 +134,47 @@ struct devInfo
   }
 };
 
+struct aieCoreInfo
+{
+  using result_type = query::aie_core_info::result_type;
+  static result_type
+  get(const xrt_core::device* device,key_type key)
+  {
+    boost::property_tree::ptree pt;
+    boost::property_tree::ptree ptarray;
+    //TODO: get max row and column for aie_metadata
+    for(int i =0;i<50;i++)
+      for(int j =0;j<8;j++)
+        ptarray.push_back(std::make_pair(std::to_string(i)+"_"+std::to_string(j), aie_sys_parser::aie_sys_read(i,(j+1),"/sys/class/aie/aiepart_0_50"))); 
+    pt.add_child("aie_core",ptarray);
+    std::ostringstream oss;
+    boost::property_tree::write_json(oss, pt);
+
+    std::string inifile_text = oss.str();
+    return inifile_text;
+  }
+};
+
+struct aieShimInfo
+{
+  using result_type = query::aie_shim_info::result_type;
+  static result_type
+  get(const xrt_core::device* device,key_type key)
+  {
+    boost::property_tree::ptree pt;
+    boost::property_tree::ptree ptarray;
+    //TODO: get max column for aie_metadata
+    for(int i=0;i<50;i++) {
```

Comment:
50 is really strange.  Please add lots of comments

Reply:
fixed